api/client/proto/joinservice.proto

```diff
@@ -19,19 +19,29 @@ import "certs.proto";
 
 import "github.com/gravitational/teleport/api/types/types.proto";
 
+message RegisterUsingIAMMethodRequest {
+    // RegisterUsingTokenRequest holds registration parameters common to all
+    // join methods.
+    types.RegisterUsingTokenRequest registerUsingTokenRequest = 1;
```

Comment:
Proto style: register_using_token_request

below: sts_identity_request.

We really to up our proto game with a formatter/linter. Since we don't have those I'll leave this at your discretion.